pkg/sbom/nodelist.go

```diff
@@ -193,13 +194,13 @@ func (nl *NodeList) MergeEdges(es []*Edge) {
 // More than one root element can be added to the NodeList.
 func (nl *NodeList) AddRootNode(n *Node) {
 	if n.Id == "" {
-		// TODO warn here
-		return
+		logrus.Warnf("Node with empty ID, creating a new Id: %s@%s\n", n.Name, n.Version)
+		n.Id = NewNodeIdentifier(fmt.Sprintf("%s@%s", n.Name, n.Version))
```

Comment:
Cool, yes autogenerate the identifier, but keep the  `auto` prefix so that the library knows it was added internally. Also drop the logrus calls, we have been trying to remove them:

```suggestion
		n.Id = NewNodeIdentifier("auto", fmt.Sprintf("%s@%s", n.Name, n.Version))
```

Reply:
got it